src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -3,20 +3,47 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.machine.Machine;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class MachineDaoImpl extends AbstractDao implements MachineDao {
-    protected MachineDaoImpl(SessionFactory sessionFactory) {
+    public MachineDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save machine " + machine.getMaker(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Machine> machineQuery = session.createQuery("FROM Machine "
+                    + "WHERE year > : age", Machine.class);
+            machineQuery.setParameter("age", age);
+            return machineQuery.getResultList();
```

Comment:
I guess year - it's production date, so the machine age will be like current_year - year (or just unclear instructions))

Reply:
Done